container/shim/src/fetchers/lassie.js

```diff
@@ -167,11 +191,36 @@ async function getRequestedBlockFromCar(streamIn, streamOut, requestedCidV1, pat
         throw new Error(`Requested CID ${requestedCidV1} doesn't equal first block CID ${rootCid}.`);
       }
 
-      sendBlockResponse(streamOut, bytes, cidV1.toString())
+      sendBlockResponse(streamOut, bytes, cidV1.toString());
     } else {
       const cacheKey = cidToCacheKey(cidV1);
       blockCache.set(cacheKey, bytes);
     }
     count++;
   }
 }
+
+async function queueMetricsReport() {
+  const date = lastMetricsReportDate;
+  const canReport = !date || new Date() - date > METRICS_REPORT_INTERVAL;
+  if (!canReport || !metrics.length) {
+    return;
+  }
+
+  const logsToReport = metrics.splice(0);
```

Comment:
```suggestion
  const logsToReport = [...metrics.splice];
```

To me this is more semantic